include/vcpkg/sourceparagraph.h

```diff
@@ -146,4 +146,13 @@ namespace vcpkg
     {
         return print_error_message({&error_info_list, 1});
     }
+
+    enum class LicenseIdCheck
+    {
+        Silent,
+        Warn,
+        Strict,
+    };
+
+    ExpectedT<std::string, msg::LocalizedString> parse_spdx_license_expression(StringView sv, LicenseIdCheck check);
```

Comment:
I'm not sure I understand this API. Why does the caller need control coupling on what happens on failure when they should be able to respond with what they want from the ExpectedT?

Reply:
(After looking at the code behind this it seems to be because it's deciding whether this going to print to stdout, but IMO this should just be returning the warning(s) rather than printing to stdout and the caller can elevate them to errors if they want)

Reply:
I'll try...